examples/seismic/acoustic/operators.py

```diff
@@ -34,30 +66,34 @@ def iso_stencil(field, m, s, damp, kernel, **kwargs):
     ----------
     field : TimeFunction
         The computed solution.
-    m : Function or float
-        Square slowness.
-    s : float or Scalar
-        The time dimension spacing.
-    damp : Function
-        The damping field for absorbing boundary condition.
-    forward : bool
-        The propagation direction. Defaults to True.
+    model : Model
+        Physical model.
+    kernel : str, optional
+        Type of discretization, 'OT2' or 'OT4'.
     q : TimeFunction, Function or float
         Full-space/time source of the wave-equation.
+    forward : bool, optional
+        Whether to propagate forward (True) or backward (False) in time.
     """
     # Forward or backward
     forward = kwargs.get('forward', True)
     # Define time step to be updated
     unext = field.forward if forward else field.backward
     udt = field.dt if forward else field.dt.T
     # Get the spacial FD
-    lap = laplacian(field, m, s, kernel)
+    lap = laplacian(field, model, kernel)
     # Get source
     q = kwargs.get('q', 0)
     # Define PDE and update rule
-    eq_time = solve(m * field.dt2 - lap - q + damp * udt, unext)
-    # Time update stencil
-    return [Eq(unext, eq_time)]
+    eq_time = solve(model.m * field.dt2 - lap - q + model.damp * udt, unext)
+
+    # Time-stepping stencil.
+    eqns = [Eq(unext, eq_time, subdomain=model.grid.subdomains['physdomain'])]
+
+    # Add free surface
+    if model.fs:
+        eqns.append(freesurface(model, Eq(unext, eq_time)))
```

Comment:
Passing `Eq(unext, eq_time)` and the splitting it up again in `freesurface`?

Reply:
Yes, this apply the free surface to the equation directly